File: 1047. Remove All Adjacent Duplicates In String/_1047.py
class Solution:
    def removeDuplicates(self, S: str) -> str:
        res = list()
        for i in S:
            if res:
                if res[-1] == i:
                    res = res[:-1]
                else:
                    res.append(i)
            else:
                res.append(i)
        return ''.join(res)

File: 1047. Remove All Adjacent Duplicates In String/1047.go
package leetcode

func removeDuplicates(S string) string {
    res := make([]string, 0)
	for _,v := range S{
		if len(res)!=0{
			if res[len(res)-1] == string(v){
				res = res[0:len(res)-1]
			}else{
				res = append(res, string(v))
			}
		}else{
			res = append(res, string(v))
		}
	}
	return strings.Join(res, "")
}